src/Raven.Client/ServerWide/DatabaseRecord.cs

```diff
@@ -489,6 +496,58 @@ public enum DeletionInProgressStatus
         HardDelete
     }
 
+    public enum MigrationStatus
+    {
+        Pending,
+        Moving,
+        Moved,
+        OwnershipTransferred
```

Comment:
Isn't this one just the same as not having a migration at all?

Reply:
we need this state, so the source will start the cleaning phase.

Reply:
I don't like that, to be honest. That is what `Moved` is for.
So the stages are:
* `BucketMigrations[bucketId].Status = Moving` - start moving data
* `BucketMigrations[bucketId].Status = Moved` - start removing data
* `BucketMigrations[bucketId]` <- missing - meaning that the process is completely done.

Any remaining will be done with the background cleanup

Reply:
I want to avoid moving the bucket again if there are still leftovers from the previous move.

Reply:
That isn't meant to be here. This is a policy decision. We'll be implementing that after, so for now that is a manual operation.

When we need it, we can have a "history" section that we'll use to remember the recent moves.

Reply:
I don't want the user to move bucket 123 from $0 to $1 and allow him to move this bucket again before the cleanup is over

Reply:
That is when it is still in `Moved` state

Reply:
Since most the part using the state isn't really wired, I will leave as it is for now and we discuss this once more code will be added.

Will add comments to each of the states